pkg/option/config.go

```diff
@@ -2892,6 +2899,7 @@ func (c *DaemonConfig) Populate() {
 	c.DisableCNPStatusUpdates = viper.GetBool(DisableCNPStatusUpdates)
 	c.EnableICMPRules = viper.GetBool(EnableICMPRules)
 	c.BypassIPAvailabilityUponRestore = viper.GetBool(BypassIPAvailabilityUponRestore)
+	c.DisableSipVerification = viper.GetBool(DisableSipVerification)
```

Comment:
I wonder about the security implications of this and how to appropriately warn users.

This can allow IP spoofing across the network via pods created with this option. This can mean bypassing security policy if someone is able to send traffic to this pod and convince it to route that traffic with an in-cluster source IP towards an in-cluster destination.

At minimum, I would be inclined to hide this option and warn very loudly and angrily to anyone deploying the option that they are potentially compromising the security of their cluster by using this option. Arguably we could perhaps discuss a more comprehensive solution to this sort of problem and come up with better guardrails. I'd welcome proposals that outline potential security concerns and mitigations with this approach.

Obviously if the user just knows best and how to properly secure their network, then they presumably could just configure this option and whatever else they need in order to have this do what they want. I don't necessarily see a reason to get in their way for this (although I could also see an argument that maybe they should just maintain these changes out-of-tree).  Mostly what I'd like to avoid is giving the impression that this is a standard supported option in Cilium that anyone can safely turn on.

Reply:
Hi, Joe. 

I understand your point, of course security should be a primary concern, I think as you suggested, the option should be hidden and if it is used, we can log an explicit warning regarding the potential security risk of running the cluster with this option set in the startup log and evaluate adding it in the cilium status command output. 

The issue I am having with source IP verification, is that it blocks all traffic exiting the pod which has a different source IP than the internal cluster IP assigned to the pod by the cluster, whether it is traffic forwarded via the pod and traffic which originated from the pod (I.E using a virtual IP). This can be problematic when running NFV applications within pods, for instance a pod can act as a VPN gateway which handles external user traffic and then forwards the traffic either internally (within the cluster) or to an external service. 

A more comprehensive solution which can be evaluated is to associate external networks (not cluster-internal) with specific endpoints, for instance, endpoint X can be configured to be associated with external networks X.X.X.X/24 and Y.Y.Y.Y/32, any traffic exiting the endpoint from these networks will be allowed, (not dropped by source IP validation mechanism).